lesson2.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * Задание 1
+ */
+function task1() {
+    let a = 1, b = 1, c, d;
+    c = ++a;        // с = 2 потому что мы сначала увеличили 'а' префиксом, потом присвоили с
+    d = b++;        // d = 1 потому что сначала присваиваем значение 'b' в 'd', а потом увеличиваем 'b' на 1
+    с = (2 + ++а); // c = 5 из-за префикса, по факту ( 2 + 1 + 2); a = 3
+    d = (2 + b++); //d = 4 из-за постфикса, по факту ( 2 + 2), потом b = (b+1) = 3
+                    //а = 3 из-за двух префиксов (1+1+1)
+                    //b = 3 из-за двух постфиксов (1+1+1)
+}
+/**
+ * Задание 2
+ * 5 т.к. по приоритету сначала выражение в скобках (а *= 2), что можно перевести как
+ * a = a * 2 = 2 * 2 = 4
+ * Соответственно x = 1 + 4 = 5
+ */
+
+/**
+ * Задание 3
+ */
+function task3() {
+    let a = 10, b = -10;
+    if (a>=0 && b>=0) {
```

Comment:
Код значительно лучше читается, если добавить пробелы между операторами и операндами
```suggestion
    if (a >= 0 && b >= 0) {
```

Reply:
Не стал отмечать все места, таких по коду много